technical_docs/Claims.md

```diff
@@ -36,19 +36,89 @@ The SciTokens project defines JWT claims specific to our problem domain.  From R
+   When rendered in JSON, the value of the `authz` claim should be a JSON list if multiple claims are present.  For example, a read-only token may have a claim of `"authz": "read"` while a read/write token may have a claim of `"authz": ["read", "write"]`.
 
-* *path* (Path): https://scitokens.org/v1/path.  The path this token is authorized to access, relative to the VOs base path on a storage system.  _NOTE_: it is understood this path should be normalized to some extent (to consider `/ligo` and `//ligo` the same location), but defining this normalization has not yet been done).
+   The `authz` claim is REQUIRED.
 
+* *site* (Site): https://scitokens.org/v1/site.  The "site name" of the service the SciToken is authorized to access.  Unlike the `aud` claim, the the site name is considered to be within the VO context, although site naming scheme may be organized by a community (such as a grid organization) or between the VO and site.  Since the `site` is within the VO context, a single service may recognize several site names.  There is no mechanism to ensure a site name is globally unique.
+
+* *path* (Path): https://scitokens.org/v1/path.  The path this token is authorized to access, relative to the VOs base path on a storage system.  When examining the contents of this attribute, paths _must_ be normalized according to [section 6 of RFC 3986](https://tools.ietf.org/html/rfc3986#section-6).  Hence, `///foo/bar/../baz` and `/foo/baz` are considered the same path for the purpose of determining access permissions.  Multiple path components may be specified as a JSON list; in such a case, the *authz* permissions are applied to all paths.  If `read` or `write` authorizations are given, then `path` MUST be specified; no default value may be assumed.
+
+SciTokens Scopes
+----------------
+
+The claims language defined by JWT is based on JSON, making it extremely flexible.  However, when creating an authorization token using the OAuth2 protocol, the client may only provide a space-delimited set of scopes; it is up to the authorization server to generate an appropriate SciToken with a list of claims.  SciTokens aims to standardize a set of scopes and the corresponding mapping to claims.
+
+* *site:NAME*.  Use this scope to request a site claim for a site `NAME` in the returned authorization token.
+* *authz:AUTHZ:PATH*.  Use this scope to request authorization against a given path.
+
+The server-side parsing of scopes should follow [Section 3.3 of RFC6749](https://tools.ietf.org/html/rfc6749#section-3.3).
+
+Care must be taken by the server in generating SciTokens in response to a request with multiple `authz` scopes.  Consider:
+
+```
+authz:read:/foo
+authz:write:/foo
+```
+
+The corresponding token claims would be:
```

Comment:
Why don't requested scopes correspond to issued scopes? I expect every OAuth access token to contain a scope per RFC 6749. Example at: https://www.oauth.com/oauth2-servers/access-tokens/self-encoded-access-tokens/